doc/gambit.txi

```diff
@@ -18110,6 +18110,23 @@ Here is an example:
 ;; (#<area-of-rectangle> #<aarea-of-rectangle> #<area-of-rectangle>) 
 @end smallexample
 
+Another basic example:
+@smallexample
+(include "~~lib/_gambit#.scm")
+
+ (##wr-set!
+    (lambda (we obj)
+      (case (macro-writeenv-style we)
+        ((mark)
+          (##default-wr we obj)) 
```

Comment:
This example does not use a good coding style.  The branch for the `mark` case must call `##wr` recursively the same way as the other case branch.  This is to ensure that the sharing of subobjects is correctly computed during the mark phase.  By calling `##default-wr` the code is assuming that the default writer will call `##wr` like the `else` branch and it is a coincidence that this is true for symbols.  The example may confuse people into thinking this works for all objects.